simple-local-avatars.dev.js

```diff
@@ -1,99 +1,332 @@
-		event.preventDefault();
-
-		if ( avatar_working )
-			return;
-
-		if ( simple_local_avatar_frame ) {
-			simple_local_avatar_frame.open();
-			return;
-		}
-
-		simple_local_avatar_frame = wp.media.frames.simple_local_avatar_frame = wp.media({
-			title: i10n_SimpleLocalAvatars.insertMediaTitle,
-			button: { text: i10n_SimpleLocalAvatars.insertIntoPost },
-			library : { type : 'image'},
-			multiple: false
-		});
-
-		simple_local_avatar_frame.on( 'select', function() {
-			// We set multiple to false so only get one image from the uploader
-			avatar_lock('lock');
-			var avatar_url = simple_local_avatar_frame.state().get('selection').first().toJSON().id;
-			jQuery.post( ajaxurl, { action: 'assign_simple_local_avatar_media', media_id: avatar_url, user_id: i10n_SimpleLocalAvatars.user_id, _wpnonce: i10n_SimpleLocalAvatars.mediaNonce }, function(data) {
-				if ( data != '' ) {
-					avatar_container.innerHTML = data;
-					$( document.getElementById('simple-local-avatar-remove') ).show();
-					avatar_ratings.disabled = false;
-					avatar_lock('unlock');
-				}
-			});
-		});
-
-		simple_local_avatar_frame.open();
-	});
-
-	$( document.getElementById('simple-local-avatar-remove') ).on('click',function(event){
-		event.preventDefault();
-
-		if ( avatar_working )
-			return;
-
-		avatar_lock('lock');
-		$.get( ajaxurl, { action: 'remove_simple_local_avatar', user_id: i10n_SimpleLocalAvatars.user_id, _wpnonce: i10n_SimpleLocalAvatars.deleteNonce })
-		.done(function(data) {
-			if ( data != '' ) {
-				avatar_container.innerHTML = data;
-				$( document.getElementById('simple-local-avatar-remove') ).hide();
-				avatar_ratings.disabled = true;
-				avatar_lock('unlock');
-			}
-		});
-	});
-
-	avatar_input.on( 'change', function( event ) {
-		avatar_preview.attr( 'srcset', '' );
-		avatar_preview.attr( 'height', 'auto' );
-		URL.revokeObjectURL( avatar_blob );
-		if ( event.target.files.length > 0 ) {
-			avatar_blob = URL.createObjectURL(event.target.files[0]);
-			avatar_preview.attr( 'src', avatar_blob );
-		} else {
-			avatar_preview.attr( 'src', current_avatar );
-		}
-	} );
-});
+let simple_local_avatar_frame;
+let avatar_spinner;
+let avatar_ratings;
+let avatar_container;
+let avatar_form_button;
+let avatar_preview;
+let avatar_input;
+let avatar_blob;
+let current_avatar;
+let avatar_working = false;
+
+jQuery( document ).ready( function ( $ ) {
+   avatar_input = $( '#simple-local-avatar' );
+   avatar_preview = $( '#simple-local-avatar-photo img' );
+   current_avatar = avatar_preview.attr( 'src' );
+   avatar_ratings = $( '#simple-local-avatar-ratings' );
+   avatar_container = $( '#simple-local-avatar-photo' );
+   $( '#simple-local-avatar-media' ).on(
+      'click',
+      function ( event ) {
+         event.preventDefault();
+
+         if ( avatar_working ) {
+            return;
+         }
+
+         /* We need to setup a Crop control that contains a few parameters
```

Comment:
Please remove comments that aren't relevant to the actual implementation - this one seems like help text from the example this came from.